hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/bootstrap/ParquetBootstrapMetadataHandler.java

```diff
@@ -62,31 +71,29 @@ Schema getAvroSchema(Path sourceFilePath) throws IOException {
   }
 
   @Override
-  void executeBootstrap(HoodieBootstrapHandle<?, ?, ?, ?> bootstrapHandle,
-                        Path sourceFilePath, KeyGeneratorInterface keyGenerator, String partitionPath, Schema avroSchema) throws Exception {
+  protected void executeBootstrap(HoodieBootstrapHandle<?, ?, ?, ?> bootstrapHandle,
+                                  Path sourceFilePath,
+                                  KeyGeneratorInterface keyGenerator,
+                                  String partitionPath,
+                                  Schema schema) throws Exception {
     BoundedInMemoryExecutor<HoodieRecord, HoodieRecord, Void> wrapper = null;
-    HoodieFileReader reader = HoodieFileReaderFactory.getReaderFactory(table.getConfig().getRecordMerger().getRecordType())
+    HoodieRecordMerger recordMerger = table.getConfig().getRecordMerger();
+
+    HoodieFileReader reader = HoodieFileReaderFactory.getReaderFactory(recordMerger.getRecordType())
             .getFileReader(table.getHadoopConf(), sourceFilePath);
     try {
+      Function<HoodieRecord, HoodieRecord> transformer = record -> {
+        String recordKey = record.getRecordKey(schema, Option.of(keyGenerator));
+        return createNewMetadataBootstrapRecord(recordKey, partitionPath, recordMerger.getRecordType())
```

Comment:
Creating `createNewMetadataBootstrapRecord` is the crux of the change here: 
 - Now metadata bootstrap record is properly initialized with schema including all of the meta-fields and not the one truncated to just record-key (HoodieSparkRecord is not able to handle such truncated meta-fields schema)
 - Avro path is restored to what it was before RFC-46